src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -394,13 +399,20 @@ impl ContainerModel {
         self.set_visibility(!self.is_active())
     }
 
-    fn set_visualization(&self, visualization:Option<visualization::Instance>) {
-        if let Some(visualization) = visualization {
-            let size = self.size.get();
-            visualization.set_size.emit(size);
-            self.view.add_child(&visualization);
-            self.visualization.replace(Some(visualization));
+    fn set_visualization
+    (&self, visualization:visualization::Instance, preprocessor:&frp::Any<enso::Code>) {
+        let size = self.size.get();
+        visualization.set_size.emit(size);
+        frp::new_network! { vis_frp_connection
+            // We need an additional "copy" node here. Doing simple
+            // `preprocessor <+ visualization.on_preprocessor_change` will not create any node in
+            // this network, so the connection will leak after setting new instance.
```

Comment:
IMO this comment is not enough. If I did not know what is happening here, I'd not understand what "leak" would mean. This comment should be clear that we are creating a new network here that will be dropped when switching visualizations, so this connection should be dropped then as well.